test/behavior/if.zig

```diff
@@ -167,3 +167,36 @@ test "if-@as-if chain" {
 
     try expect(num_frames == 4);
 }
+
+fn returnTrue() bool {
+    return true;
+}
+
+test "if value shouldn't be load-elided if used later (structs)" {
+    const Foo = struct { x: i32 };
+
+    var a = Foo{ .x = 1 };
+    var b = Foo{ .x = 1 };
+
+    const c = if (@call(.never_inline, returnTrue, .{})) a else b;
```

Comment:
`never_inline` is not able to stop LLVM optimizations so maybe it'd be better to stick to using a `var` like in the issue.

Reply:
But as I understand, LLVM optimizations do not matter in this issue: the zig codegen had been making a wrong decision about load-elision (and consequently even `-O ReleaseFast` made `c.x` equal to 2).

I've tried to make the test somewhat future-proof relative to the possible changes in the zig codegen (e.g. if it started to do some basic optimizations before producing IR, the test would no longer be relevant with a simple `var cond = false; cond = true;`)